File: src/code/python/matrix/diagonals.py
def fn(matrix: list[list[int]]):
    r = len(matrix)
    c = len(matrix[0])

    main_diagonal = [matrix[i][i] for i in range(min(r, c))]
    anti_diagonal = [matrix[i][~i] for i in range(min(r, c))]

File: src/code/ruby/matrix/diagonals.rb
def fn(matrix)
  r = matrix.size
  c = matrix[0].size

  main_diagonal = (0...[r, c].min).map { |i| matrix[i][i] }
  anti_diagonal = (0...[r, c].min).map { |i| matrix[i][-i - 1] }
end
